Qt.py

```diff
@@ -761,22 +762,30 @@ def _pyqt5():
 
 def _pyqt4():
     """Initialise PyQt4"""
-
+    
     import sip
-    try:
-        sip.setapi("QString", 2)
-        sip.setapi("QVariant", 2)
-        sip.setapi("QDate", 2)
-        sip.setapi("QDateTime", 2)
-        sip.setapi("QTextStream", 2)
-        sip.setapi("QTime", 2)
-        sip.setapi("QUrl", 2)
-    except AttributeError as e:
-        raise ImportError(str(e))
-        # PyQt4 < v4.6
-    except ValueError as e:
-        # API version already set to v1
-        raise ImportError(str(e))
+
+    # Validation of env var. Prevents an error if the env var is invalid since it's just a hint.
+    api_hint = 2
+    if QT_SIP_API_HINT in ('1', '2'):
+        api_hint = int(QT_SIP_API_HINT)
+    elif QT_SIP_API_HINT:
+        _log("Warning: QT_SIP_API_HINT is not a valid value of 1 or 2: [%s]" % (QT_SIP_API_HINT))
+
+    api_names = ('QString', 'QVariant', 'QDate', 'QDateTime', 'QTextStream', 'QTime', 'QUrl')
+
+    for api_name in api_names:
+        try:
+            sip.setapi(api_name, api_hint)
+        except AttributeError as e:
+            raise ImportError(str(e))
+            # PyQt4 < v4.6
+        except ValueError as e:
+            # API version already set to a different version
+            if QT_SIP_API_HINT:
+                _log("Warning: API '%s' has already been set to %s" % (api_name, sip.getapi(api_name)))
```

Comment:
Yeah I agree it's not the prettiest but I can't think of a good way to consolidate it further.
As for printing always instead of logging, I think you're right. It is an important enough message that it shouldn't be invisible.

I'll make that change later today.